From stable+bounces-114133-greg=kroah.com@vger.kernel.org Thu Feb  6 17:22:09 2025
From: Koichiro Den <koichiro.den@canonical.com>
Date: Fri,  7 Feb 2025 01:21:30 +0900
Subject: Revert "btrfs: avoid monopolizing a core when activating a swap file"
To: gregkh@linuxfoundation.org, stable@vger.kernel.org
Cc: wqu@suse.com, fdmanana@suse.com, dsterba@suse.com
Message-ID: <20250206162131.1387235-1-koichiro.den@canonical.com>

From: Koichiro Den <koichiro.den@canonical.com>

This reverts commit 6e1a8225930719a9f352d56320214e33e2dde0a6.

The backport for linux-6.6.y, commit 6e1a82259307 ("btrfs: avoid
monopolizing a core when activating a swap file"), inserted
cond_resched() in the wrong location.

Revert it now; a subsequent commit will re-backport the original patch.

Fixes: 6e1a82259307 ("btrfs: avoid monopolizing a core when activating a swap file") # linux-6.6.y
Signed-off-by: Koichiro Den <koichiro.den@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 fs/btrfs/inode.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -7153,8 +7153,6 @@ noinline int can_nocow_extent(struct ino
 			ret = -EAGAIN;
 			goto out;
 		}
-
-		cond_resched();
 	}
 
 	if (orig_start)
